src/features/Search/movieSlice.js

```diff
@@ -0,0 +1,47 @@
+import { createSlice } from "@reduxjs/toolkit";
+
+const movieSlice = createSlice({
+  name: "movie",
+  initialState: {
+    loading: true,
+    error: false,
+    movie: null,
+    movieCredits: null,
+  },
+
+  reducers: {
+    fetchMovieSuccess: (state, { payload: movie }) => {
+      state.loading = false;
+      state.error = false;
+      state.movie = movie;
+    },
+    fetchMovieError: (state) => {
+      state.error = true;
+    },
+    fetchMovie: (state) => {
+      state.loading = true;
+    },
+    fetchMovieCreditsSuccess: (state, { payload: movieCredits }) => {
+      state.loading = false;
+      state.error = false;
+      state.movieCredits = movieCredits;
+    },
+  },
+});
+
+export const {
+  fetchMovieSuccess,
+  fetchMovieError,
+  fetchMovie,
+  fetchMovieCreditsSuccess,
+} = movieSlice.actions;
+
+const selectMovieState = (state) => state.movie;
```

Comment:
Looks good but what do you think if we named this code differently like "const selectSearchState"?